src/Microsoft.ML.Core/Data/ReadOnlyMemoryUtils.cs

```diff
@@ -0,0 +1,285 @@
+
+            if (index == -1)
+            {
+                left = memory;
+                right = default;
+                return false;
+            }
+
+            left = memory.Slice(0, index);
+            right = memory.Slice(index + 1, memory.Length - index - 1);
+            return true;
+        }
+
+        /// <summary>
+        /// Returns a <see cref="ReadOnlyMemory{T}"/> of <see cref="char"/> with leading and trailing spaces trimmed. Note that this
+        /// will remove only spaces, not any form of whitespace.
+        /// </summary>
+        public static ReadOnlyMemory<char> TrimSpaces(ReadOnlyMemory<char> memory)
+        {
+            if (memory.IsEmpty)
+                return memory;
+
+            int ichLim = memory.Length;
+            int ichMin = 0;
+            var span = memory.Span;
+            if (span[ichMin] != ' ' && span[ichLim - 1] != ' ')
+                return memory;
+
+            while (ichMin < ichLim && span[ichMin] == ' ')
+                ichMin++;
+            while (ichMin < ichLim && span[ichLim - 1] == ' ')
+                ichLim--;
+            return memory.Slice(ichMin, ichLim - ichMin);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="ReadOnlyMemory{T}"/> of <see cref="char"/> with leading and trailing whitespace trimmed.
+        /// </summary>
+        public static ReadOnlyMemory<char> TrimWhiteSpace(ReadOnlyMemory<char> memory)
+        {
+            if (memory.IsEmpty)
+                return memory;
+
+            int ichMin = 0;
+            int ichLim = memory.Length;
+            var span = memory.Span;
+            if (!char.IsWhiteSpace(span[ichMin]) && !char.IsWhiteSpace(span[ichLim - 1]))
+                return memory;
+
+            while (ichMin < ichLim && char.IsWhiteSpace(span[ichMin]))
+                ichMin++;
+            while (ichMin < ichLim && char.IsWhiteSpace(span[ichLim - 1]))
+                ichLim--;
+
+            return memory.Slice(ichMin, ichLim - ichMin);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="ReadOnlyMemory{T}"/> of <see cref="char"/> with trailing whitespace trimmed.
+        /// </summary>
+        public static ReadOnlyMemory<char> TrimEndWhiteSpace(ReadOnlyMemory<char> memory)
+        {
+            if (memory.IsEmpty)
+                return memory;
+
+            int ichLim = memory.Length;
+            var span = memory.Span;
+            if (!char.IsWhiteSpace(span[ichLim - 1]))
+                return memory;
+
+            while (0 < ichLim && char.IsWhiteSpace(span[ichLim - 1]))
+                ichLim--;
+
+            return memory.Slice(0, ichLim);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="ReadOnlyMemory{T}"/> of <see cref="char"/> with trailing whitespace trimmed.
+        /// </summary>
+        public static ReadOnlyMemory<char> TrimEndWhiteSpace(ReadOnlyMemory<char> memory, ReadOnlySpan<char> span)
```

Comment:
I think this method is a little awkward and should be removed. It is assuming that the `memory` and `span` are pointing to the same location. I see it is only called in 1 spot, which can just be moved to the above overload that only takes a `ROM<char>`.